src/common/buffer.cc

```diff
@@ -800,53 +836,175 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
 
   bool buffer::ptr::is_zero() const
   {
-    const char *data = c_str();
-    for (size_t p = 0; p < _len; p++) {
-      if (data[p] != 0) {
-	return false;
-      }
+    const char* data = c_str();
+    const char* max = data + _len;
+    const char* max32 = data + (_len / sizeof(uint32_t))*sizeof(uint32_t);
+#if defined(__GNUC__) && defined(__x86_64__)
+    // we do have XMM registers in x86-64, so if we need to check at least
+    // 16 bytes, make use of them 
+    int left = _len;
+    if (left / sizeof(uint128_t) > 0) {
+        // align data pointer to 16 bytes, otherwise it'll segfault due to bug
+        // in (at least some) GCC versions (using MOVAPS instead of MOVUPS).
+        // check up to 15 first bytes while at it.
```

Comment:
On x86-64, for uint128_t it assumes that it's 16-byte aligned, so it uses MOVAPS (move aligned packed single) instruction instead of MOVUPS (move unaligned packed single), causing program to segfault when data are actually unaligned. And AFAIK you can't force it to use MOVUPS.